Parsers/ASimNetworkSession/Parsers/vimNetworkSessionWatchGuardFirewareOS.yaml

```diff
@@ -0,0 +1,196 @@
+ParserName: vimNetworkSessionWatchGuardFirewareOS                                  
+EquivalentBuiltInParser: _Im_NetworkSession_WatchGuardFirewareOS                                  
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstportnumber
+    Type: int
+    Default: int(null)
+  - Name: hostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dvcaction
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let Parser=(starttime:datetime=datetime(null), endtime:datetime=datetime(null), srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]), dstportnumber:int=int(null), hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]), eventresult:string='*', disabled:bool=false){
+    let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+    let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+    let ip_any = set_union(srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+    let EventLookup=datatable(DvcAction:string,EventResult:string,EventSeverity:string)
+    [
+        "Allow","Success","Informational"
+        , "Deny","Failure","Low"
+    ];
+    let SyslogParser = (Syslog:(SyslogMessage:string)) {
+      Syslog
+      | parse-kv SyslogMessage as (geo_src:string
+      , geo_dst:string
+      , src_user:string
+      , dst_user:string
+      , duration:int
+      , sent_bytes:long
+      , rcvd_bytes:long) with (pair_delimiter=' ', kv_delimiter='=', quote='"')
+      | project-rename SrcGeoCountry = geo_src
+      , DstGeoCountry = geo_dst
+      , SrcUsername = src_user
+      , DstUsername = dst_user
+      , NetworkDuration = duration
+      , SrcBytes = sent_bytes
+      , DstBytes = rcvd_bytes
+      | extend SrcUsernameType = iif(SrcUsername has "@","UPN","")
+      | extend DstUsernameType = iif(DstUsername has "@","UPN","")
+      | parse SyslogMessage with * "repeated " EventCount:int " times" *
+      | extend EventCount = iif(isnotempty(EventCount), EventCount, toint(1))
+      | project-away SyslogMessage
+    };
+    let IPParser = (Syslog:(SrcIpAddr:string,DstIpAddr:string)){
+      Syslog
+      | extend temp_SrcMatch = has_any_ipv4_prefix(SrcIpAddr,src_or_any)
+      , temp_DstMatch = has_any_ipv4_prefix(DstIpAddr,dst_or_any)
+      | extend ASimMatchingIpAddr = case(
```

Comment:
Did you test the filters? It has to be manual and is very error prone. We can discuss tomorrow how to best do that.

Reply:
Filters are tested and work.